src/backend/common/CommandBuffer.cpp

```diff
@@ -201,6 +201,19 @@ namespace backend {
             auto it = mostRecentTextureUsages.find(texture);
             return it != mostRecentTextureUsages.end() && (it->second & usage);
         };
+        auto isTextureTransitionPossible = [&](TextureBase* texture, nxt::TextureUsageBit usage) -> bool {
+            const nxt::TextureUsageBit attachmentUsages =
+                nxt::TextureUsageBit::ColorAttachment |
+                nxt::TextureUsageBit::DepthStencilAttachment;
+            ASSERT(usage != nxt::TextureUsageBit::None && nxt::HasZeroOrOneBits(usage));
+            auto it = mostRecentTextureUsages.find(texture);
```

Comment:
This checks makes it so the following produces an error, inside our outside of a renderpass:

```cpp
cmdBufBuilder->TransitionTextureUsage(tex, nxt::TextureUsageBit::ColorAttachment);
cmdBufBuilder->TransitionTextureUsage(tex, nxt::TextureUsageBit::Sampled);
```

This is surprising behavior, I think it would be simpler to specify that the texture is "temporarily" frozen when it is a (color / DS / input) attachment for the current subpass.